packages/angular_devkit/schematics/src/rules/base.ts

```diff
@@ -106,7 +106,10 @@ export function branchAndMerge(rule: Rule, strategy = MergeStrategy.Default): Ru
     const branchedTree = branch(tree);
 
     return callRule(rule, observableOf(branchedTree), context)
-      .pipe(map(t => staticMerge(tree, t, strategy)));
+      .pipe(
+        last(),
```

Comment:
`last` throws if there are no elements.  As long as `callRule` guarantees at least one element, this should be fine.

Reply:
All our observables should be containing a value.